bluecellulab/cell/core.py

```diff
@@ -254,34 +254,21 @@ def get_psection(self, section_id=None, secname=None):
             raise Exception(
                 "Cell: get_psection requires or a section_id or a secname")
 
-    def get_hsection(self, section_id):
-        """Use the serialized object to find a hoc section from a section id.
-
-        Parameters
-        ----------
-        section_id : int
-                    Section id
-
-        Returns
-        -------
-        hsection : nrnSection
-                   The requested hoc section
-        """
-
+    def get_hsection(self, section_id: int | float) -> NeuronSection:
+        """Use the serialized object to find a hoc section from a section
+        id."""
+        section_id = int(section_id)
         # section are not serialized yet, do it now
         if self.serialized is None:
             self.serialized = SerializedSections(self.cell.getCell())
 
         try:
-            sec_ref = self.serialized.isec2sec[int(section_id)]
+            sec_ref = self.serialized.isec2sec[section_id]
         except IndexError as e:
             raise IndexError(
-                "bluecellulab get_hsection: section-id %s not found in %s" %
-                (section_id, self.morphology_path)) from e
-        if sec_ref is not None:
```

Comment:
Is there a possibility that sec_ref is None here?

Reply:
Ah yes, I noticed that variable is of type `neuron.h.SectionRef` and it can never be None (it'll be a HocObject type) unless we explicitly set it to be None (which we do not do).
https://www.neuron.yale.edu/neuron/static/py_doc/modelspec/programmatic/topology/secref.html